src/module/order/application/service/order.service.ts

```diff
@@ -56,11 +84,27 @@ export class OrderService {
     }
     const order = user.orders.find((order) => order.id === orderId);
     if (!order) {
-      throw new NotFoundException();
+      throw new CannotAccessOrderException()
     }
     if (order.status !== OrderStatus.pending) {
       throw new CannotUpdateOrderException();
     }
     return await this.orderRepository.update(orderId, updatedOrder);
   }
+
+  async saveItems(items: Item[], savedOrder: Order): Promise<Item[]> {
+    const savedItems = items.map(async (item: Item) => {
+      const newItem = new Item();
+      const product = await this.productService.getOne(item.product.id);
+      const subtotal = product.price * item.quantity;
+      newItem.product = product;
+      newItem.order = savedOrder;
+      newItem.subTotal = subtotal;
+      newItem.quantity = item.quantity;
+      const savedItem = await this.itemService.save(newItem);
```

Comment:
i guess we should make a repository method to save in a bulk. it can be the same save method but if is an array should save the array.

Reply:
ok